src/uglify/index.js

```diff
@@ -0,0 +1,83 @@
+import os from 'os';
+import findCacheDir from 'find-cache-dir';
+import workerFarm from 'worker-farm';
+import { get, put } from './cache';
+import { encode } from './serialization';
+import versions from './versions';
+import worker from './worker';
+
+let workerFile = require.resolve('./worker');
+
+try {
+  // run test
+  workerFile = require.resolve('../../dist/uglify/worker');
+} catch (e) { } // eslint-disable-line no-empty
+
+class Uglify {
+  constructor({
+    cache = findCacheDir({ name: 'uglifyjs-webpack-plugin' }),
+    workers = os.cpus().length - 1,
```

Comment:
This still doesn't prevent a user from passing in a really high number of workers. Which I guess is ok?